### ISSUE TITLE
Phase09-Part2-Team3

### ISSUE THREAD

**Issue by kimianoorbakhsh** [Star-Academy/codestar-intern-issues#98]
This issue template is based of [Phase09-Web.md](https://github.com/Star-Academy/codestar-internship/blob/master/Projects/Phase09-Web/Phase09-Web.md) contents.
Click on "Raw" button on right side of this page to see original markdown then create a new issue with orginal content of this tasklist:

# Phase 9 Second Part

## UI/UX

- [x] Read UI/UX part of document and answer questions below:
  - [x] What is the difference between UI and UX?
  
    `At the most basic level, UI is made up of all the elements that enable someone to interact with a product or service. UX, on the other hand, is what the individual interacting with that product or service takes away from the entire experience.`
  - [x] Which side of brain is for UI?
  
    `Right side`
  - [x] What is a breadcrumb?
  
    `A “breadcrumb” (or “breadcrumb trail”) is a type of secondary navigation scheme that reveals the user’s location in a website or Web application`
  - [x] Play at least easy part of can't unsee game 
- [x] Design a simple website for your current console-based search engine.

## HTML

- [x] Read HTML part of document and answer questions below:
  - [x] What does `<!DOCTYPE html>` means?
  
    `Tells the browser that the file must be read as HTML5`
  - [x] Write following text inside paragraph using HTML.
  this is <b>bold</b> and this is <i>italic</i> text.
  
    <P>this is <b>bold</b> and this is <i>italic</i> text.</P>
  - [x] Write a simple table with 2 cell and 2 row, using HTML. (use <a target="_blank" href="https://www.w3schools.com/tags/">this link</a> for help)
  
    <table>
<tr> 
<th>
:)
</th>
</tr>
<tr> 
<td>
=)
</td>
</tr>
     </table>
  - [ ] Write a form with following fields. Name which is string, gender which is a radio button, age which is number and a check box for reading the privacy policy. 
  
```
<form>
  <label><h5>Name : &ensp;
  <input type="text" name="fname"  placeholder="Name">     </h5></lable>
  
  <h5>Male  
  <input type="radio" name="male" >
  &ensp; &ensp;Famale  
  <input type="radio" name="male">
  </h5>

  <label><h5>Age : &ensp;
  <input type="number" name="age">
  </h5></label> 
  
  <label><h5>I confirm I have read and accept privacy     policy
  <input type="checkbox" name="age">
  </h5></label> 
</form>
```

- [x] Implement your search engine website using only HTML. It has static content and does not connect to server. (You need two page one for home page and another for result page like google)

## CSS

- [x] Read CSS part of document and answer questions below:
  - [x] Why do we need css?
    
      `To make the pages more attractive and beautiful and also to make access easier`
      
  - [x] What is the syntax for linking a css file with the name of myStyle.css to html page.
  
    ` <link rel="stylesheet" href="./myStyle.css" />`
  - [x] What would following selector selects.(check all the correct answers) input.name-field :
    - [ ] `<input type="text" />`

    - [ ] `<span class="name-field"></span>`

    - [ ] `<span class="name-field"><input type="text" /></span>`

    - [x] `<input type="number" class="name-field" />`

    - [x] `<input type="text" class="name-field" />`

    - [ ] `<input type="text" name="name-field" />`
  - [x] What css property we can use, to make our menu appear always on top even when we scrolling to bottom of the page. (use <a target="_blank" href="https://www.w3schools.com/css/css_positioning.asp">this link</a> for help)
  
    `position and it s value should be fixed `
  - [x] What is a block display and by defalut which display does a span tag has?
  
    `Block display takes up all the space, regardless of the amount of space required and by default display of span tag is "inline"`
- [x] Add css styles to your search engine pure html website.

## Javascript

- [x] Read Javascript part of document and answer questions below:
  - [x] Why we need javascript?
    
      `To implement logic of websites we need javascript`
      
  - [x] What is the syntax for loading a javascript file with the name of myCode.js to html page.
  
    `<script src="./myCode.js"></script>`
  - [x] Write a function to log numbers between 1 - 100 using javascript:

    ```
    function calculateLogs() {
        let logs[];
        let i ;
        for(i=1 ; i<=100 ; i++)
           logs.push(Math.log(i));
      }
     ```
  - [x] What is AJAX request?
  
    `AJAX is a technique for creating fast and dynamic web pages. AJAX allows web pages to be updated asynchronously by exchanging small amounts of data with the server behind the scenes. This means that it is possible to update parts of a web page, without `
  - [x] Write the code for sending an AJAX post request to server. The server will get name and age as parameters.
  
        
        const xhttp = new XMLHttpRequest();
        xhttp.open('POST', 'https://localhost:4200/sky/find');
        xhttp.send("name=sky&age=0");
        
- [x] Connect your search engine website to server api and fill the html content with your server result.

## Angular

- [x] Read Angular part of document and answer questions below:
  - [x] What is Angular?
    
      `Angular is a platform and framework for building single-page client applications using HTML and TypeScript. Angular is written in TypeScript. It implements core and optional functionality as a set of TypeScript libraries that you import into your apps.`
      
  - [x] Write at least three difference between typescript and javascript.
  
    `1- TypesScript is known as Object oriented programming language whereas JavaScript is a scripting language.
2- TypeScript has a feature which is strongly-typed or supports static typing. That means Static typing allows for checking type correctness at compile time. This is not available in JavaScript.
3- TypeScript has Interface but JavaScript does not have Interface.`
  - [x] Install Angular.
  - [x] Generate new project.
    - [x] Use routing. 
    - [x] Use scss for styles.
    - [x] Add Angular material to your project.
    - [x] Generate two component for your search website. (one for first page and another for your result page)
    - [x] Add your component to routing.
    - [x] use your first page as default page when your website is visited.
  - [x] Run project.
  - [x] What is a SPA application?
  
    `A single-page application (SPA) is a web application or website that interacts with the web browser by dynamically rewriting the current web page with new data from the web server, instead of the default method of the browser loading entire new pages. The goal is faster transitions that make the website feel more like a native app.`
  - [x] What is a service and how do we provide them?
  
    `Angular services are singleton objects that get instantiated only once during the lifetime of an application. They contain methods that maintain data throughout the life of an application, i.e. data does not get refreshed and is available all the time.
<br>By using @Injectable decorator you will tell that other classes , services can have this service as a dependency`
- [x] Rewrite your search website code and use Angular for your website.

# Review
Link of your own PR:

[Our PR](https://github.com/Star-Academy/Team3-Codes/pull/9)

Link of other PR(s) you agreed to review:

[Team 4](https://github.com/Star-Academy/Team4-Codes/pull/8)

- [x] Your own PR is reviewed and approved by at least one other team.

- [ ] Your own PR is reviewed and approved by mentor (hamyar).

- [x] Your own PR is reviewed and approved by manager (rahbar).

- [x] Your own PR is merged.
